tests/unittest/llmapi/test_kv_cache_dtype_override.py

```
@@ -0,0 +1,66 @@
+
+def test_get_llm_args_plumbs_kv_cache_dtype():
+    llm_args, _ = get_llm_args(model="dummy", kv_cache_dtype="nvfp4")
+    assert llm_args["kv_cache_config"].dtype == "nvfp4"
+
+
+def test_kv_cache_config_dtype_validation():
+    cfg = KvCacheConfig(dtype="NVFP4")
+    assert cfg.dtype == "nvfp4"
+
+    with pytest.raises(ValueError,
+                       match="kv_cache_config.dtype must be one of"):
```

Ruff (0.15.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)